File: 2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1/2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minOperations = function(nums) {
  const n = nums.length;
  const ones = nums.reduce((acc, num) => acc + +(num === 1), 0);
  
  if (ones) return n - ones;
  
  let diff = n;
  
  for (let i = 0; i < n; i++) {
    let g = nums[i];
    for (let j = i + 1; j < n; j++) {
      g = gcd(g, nums[j]);
      if (g === 1) {
        diff = Math.min(diff, j - i);
        break;
      }
    }
  }
  
  return diff === n ? -1 : diff + n - 1;
};

const gcd = (a, b) => (!b ? a : gcd(b, a % b));

File: 2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1/2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        ones = nums.count(1)
        if ones:
            return n - ones
        diff = n
        for i in range(n):
            g = nums[i]
            for j in range(i + 1, n):
                g = gcd(g, nums[j])
                if g == 1:
                    diff = min(diff, j - i)
                    break
        return -1 if diff == n else diff + n - 1